dali/pipeline/operators/sequence/sequence_rearrange.cu

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+#include <vector>
+
+#include "dali/pipeline/operators/sequence/sequence_rearrange.h"
+
+namespace dali {
+
+template <>
+void SequenceRearrange<GPUBackend>::RunImpl(DeviceWorkspace *ws, const int idx) {
```

Comment:
Can you extract the common part of GPU and CPU implementation and just specialize the `for` loop?